ethereum/contracts.go

```diff
@@ -22,4 +22,50 @@ type BridgeContract interface {
 		recipient []common.Address,
 		token []common.Address,
 	) (AssetsLockedIterator, error)
+
+	PastAssetsUnlockConfirmedEvents(
```

Comment:
Shouldn't this just be:
```suggestion
	PastAssetsUnlockEvents(
```
How this method is supposed to ensure the events are confirmed? Isn't it about taking the right block range (i.e. finalized) on Ethereum?

Reply:
We cannot name this method `PastAssetsUnlockEvents`. We need to keep it as `PastAssetsUnlockConfirmedEvents` as this is the name that will be generated when the package for MezoBridgeV2 is released. Notice that all the functions in this interface are simply functions in the generated bindings.

> How this method is supposed to ensure the events are confirmed?
This method does not guarantee anything on its own, it's a pure connection to the MezoBridge contract that returns events based on the provided filters. It's up to us to call it with the correct filters. If we want only finalized events, we need to set its `endBlock` field with the finalized block number (64-96 blocks behind the current tip).

Reply:
> Why it will be generated with this name?
Names of these functions are generated in bindings according to names of events in the smart contract, e.g.:
for 
```
event AssetsLocked {...}
```
we have 
```
func (mb *MezoBridge) PastAssetsLockedEvents(
	startBlock uint64,
	endBlock *uint64,
	sequenceNumberFilter []*big.Int,
	recipientFilter []common.Address,
	tokenFilter []common.Address,
) ([]*abi.MezoBridgeAssetsLocked, error) {..}
```

Reply:
oh yeah, you are totally right. I had a brain fog and totally missed the fact the name of the event itself is `AssetsUnlockConfirmed` 🤦 Nevermind. Ignore my original comment please.